File: 141. Linked List Cycle/solution.js
var hasCycle = function(head) {
    if (head === null) {
        return false;
    }

    let first = head;
    let second = head.next;

    while (second !== null && second.next !== null) {
        first = first.next;
        second = second.next.next;

        if (first === second) {
            return true;
        }
    }
    
    return false;
};

File: 141. Linked List Cycle/solution.go
package main

func hasCycle(head *ListNode) bool {
	if head == nil {
		return false
	}

	first := head
	second := head.Next

	for second != nil && second.Next != nil {
		first = first.Next
		second = second.Next.Next

		if first == second {
			return true
		}
	}

	return false
}
